src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,98 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] items;
+
+    public ArrayList() {
+        items = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == items.length) {
+            resize();
+        }
+        items[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndexForAdd(index);
+        if (size == items.length) {
+            resize();
+        }
+        System.arraycopy(items, index, items, index + 1, size++ - index);
+        items[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return items[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        items[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        T deletedObject = items[index];
+        System.arraycopy(items, index + 1, items, index, --size - index);
+        return deletedObject;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < size; i++) {
+            if (Objects.equals(items[i], element)) {
```

Comment:
Don't use Objects, Arrays, or any other util class.

Reply:
Intelji advises doing this
![foto1](https://github.com/mate-academy/jv-array-list/assets/127691880/123a9ddb-344d-4113-a44e-db0510fc5666)